scope.go

```diff
@@ -66,11 +66,23 @@ func (s *Scope) newSource(name string, constructor func() StatSource) (
 	return ss
 }
 
-// FuncNamed retrieves or creates a Func named after the given name. See
-// Func() for automatic name determination.
-func (s *Scope) FuncNamed(name string) *Func {
-	source := s.newSource("func:"+name, func() StatSource {
-		return newFunc(s, NewSeriesKey("function").WithTag("name", name))
+// FuncNamed retrieves or creates a Func named using the given name and tags.
+// See Func() for automatic name determination.
+//
+// Each unique combination of tags (key/value) will result in a unique Func.
+// Tags are not sorted, so keep the order consistent to avoid creating new
+// Funcs.
+func (s *Scope) FuncNamed(name string, tags ...*Tag) *Func {
+	sourceName := "func:" + name
+	for _, tag := range tags {
+		sourceName = sourceName + tag.Key + tag.Val
```

Comment:
Actually I just realized that if you do `foo=bar` and `foob=ar` then you get the same key. We should should do `sourceName + "," + tag.Key + "=" + tag.Val` and perhaps change it to use like `strings.Builder` to avoid tons of allocations and quadratic time.